reactive/kotlinx-coroutines-rx2/src/RxAwait.kt

```diff
@@ -41,7 +39,10 @@ public suspend fun CompletableSource.await(): Unit = suspendCancellableCoroutine
  * immediately resumes with [CancellationException].
  */
 @Suppress("UNCHECKED_CAST")
-public suspend fun <T> MaybeSource<T>.await(): T? = (this as MaybeSource<T?>).awaitOrDefault(null)
+public suspend fun <T> MaybeSource<T>.await(): T? {
+    val result = (this as MaybeSource<Any>).awaitOrDefault(MISSING_VALUE)
+    return if (result === MISSING_VALUE) null else result as T
```

Comment:
What's the purpose of this change? `MaybeSource` is supposed to be null-hostile and cannot return produce `null` value.

Reply:
Yes, but because of its nature, it can emit nothing and just complete. In such a situation, we return `null`.